### DISCUSSION TITLE
Enabling above 4G decoding not found with IFR Extractor

### DISCUSSION THREAD

**Discussion by Pastafazou** [xCuri0/ReBarUEFI#52] [Q&A] [answered]
Hi,

I was trying to modify the BIOS to enable above 4G decoding on my Dell Optiplex 9020 (Dell 0PC5F7 A02 Motherboard).
As I understand it, the chipset should be able to allow above 4G decoding despite not officially supporting resizable Bar, being equipped with a Haswell chip (i7-4770)
After extracting the BIOS image and opening it in UEFITool, I was able to find the following:

Unicode text "4G Decod" found in Setup/.../97E409E6-4CC1-11D9-81F6-000000000000 at header-offset 3C6F7h
Unicode text "Above 4G" found in Setup/.../97E409E6-4CC1-11D9-81F6-000000000000 at header-offset 1198Fh
Unicode text "Above 4G" found in Setup/.../97E409E6-4CC1-11D9-81F6-000000000000 at header-offset 3C6EBh
Unicode text "Above 4G" found in Setup/.../97E409E6-4CC1-11D9-81F6-000000000000 at header-offset 3C786h

As such, the setting should be found within the BIOS image. However, extracting 97E409E6-4CC1-11D9-81F6-000000000000 (where UEFITool indicates the setting should be located) using the provided IFR extractor (IRFExtractor_0.3.6_win) yields a text file which doesn't contain either '4G Decod' or 'Above 4G'.

I'm still hoping this isn't a dead end, but even the headers provided by UEFITool seem to be in the wrong format.
I know this repository is primarily used to enable resizable bar, but I haven't been able to find any information anywhere else.

Any help would be greatly appreciated,
Pierre

P.S. I have attached a zipped copy of both the extracted BIOS image and IFR file (As github doesn't allow .BIN files), in case someone with more technical eyes than mine wanted to take a look.
[Extracted_BIOS_IFR.zip](https://github.com/xCuri0/ReBarUEFI/files/11505018/Extracted_BIOS_IFR.zip)

**Comment by Pastafazou** [comment]
Just an update to this problem, 

Looking at the hex view at the headers found with UEFITool, I was able to find the string "Enable or disable memory remap above 4G" (see attached image) which leads me to believe my issue is different from the one encountered in [this](https://github.com/xCuri0/ReBarUEFI/issues/14) issue, as the setting description, like the one he has, can _also_ be found at header offsets 3C6E-3C78 (string stating "Above 4G Decoding. Enables or Disables 64 bit capable devices to be decoded in Above 4G Address Space").

![tool](https://github.com/xCuri0/ReBarUEFI/assets/133947223/5d9b812e-b2a8-4e98-9cfb-2823d39d695a)

EDIT: after [this](https://winraid.level1techs.com/t/success-above-4g-decoding-and-resizable-bar-bios-mod-for-asus-p8z77-v-pro/90465/3?page=2) blog post i've realized the option refers to physical memory and has nothing to do with decoding

Could it be that enabling the option may require DSDT modification? If so, does anyone have any pointers as where to start? I know this would be a quite complex task to enable an option on a fairly weak system, but i'm willing to put in the work.

Any help would be greatly appreciated!

**Comment by xCuri0** [comment, marked as answer]
Duplicate of #14, yes you can do the DSDT modification (follow the wiki) and that will get it working in Linux only
